lib/coupon_code.rb

```diff
@@ -21,8 +21,8 @@ def self.generate(options = { parts: PARTS })
 
   def self.validate(orig, num_parts = PARTS)
     code = orig.upcase
-    code.gsub!(/[^0-9A-Z]+/, '')
-    parts = code.scan(/[0-9A-Z]{#{LENGTH}}/)
+    code.gsub!(reg_symbol, '')
```

Comment:
Hi @baxang, I was trying _not_ to add more lines to `self.validate` as it's pretty crowded on it's current state.